VM stuck Scheduled on btrfs: failed to detect root mount info of containerDisk

> /kind bug

**What happened**:

Started VM, but it didn't start or cause an error.
As there doesn't seem to be good googleable error or any interesting log message I'm kinda stuck, any clue?

**What you expected to happen**:

An actual running VM or an error.

**How to reproduce it (as minimally and precisely as possible)**:

* Install kubernetes 1.16.3 using RKE
* Install kubevirt v0.26.1
* Start a VM.

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): v0.26.1
- Kubernetes version (use `kubectl version`): v1.16.3
- Cloud provider or hardware configuration: bare metal HP Proliant DL360
- OS (e.g. from /etc/os-release): 18.04.4 LTS
- Kernel (e.g. `uname -a`):
- Install tools: RKE

- VM or VMI specifications: 

```
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: debian
spec:
  running: false
  template:
    metadata:
      labels: 
        kubevirt.io/size: small
        kubevirt.io/domain: debian
    spec:
      domain:
        cpu:
          cores: 2
        devices:
          disks:
            - name: containervolume
              disk:
                bus: virtio
            - name: cloudinitvolume
              disk:
                bus: virtio
          interfaces:
          - name: default
            bridge: {}
        resources:
          requests:
            memory: 1024M
      networks:
      - name: default
        pod: {}
      volumes:
        - name: containervolume
          containerDisk:
            image: tedezed/debian-container-disk:9.0
        - name: cloudinitvolume
          cloudInitNoCloud:
            userData: |-
              #cloud-config
              chpasswd:
                list: |
                  debian:debian
                  root:toor
                expire: FalseapiVersion: kubevirt.io/v1alpha3
```

```
# kubectl describe virtualmachine.kubevirt.io/debian
Name:         debian
Namespace:    flfr
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"kubevirt.io/v1alpha3","kind":"VirtualMachine","metadata":{"annotations":{},"name":"debian","namespace":"flfr"},"spec":{"run...
              kubevirt.io/latest-observed-api-version: v1alpha3
              kubevirt.io/storage-observed-api-version: v1alpha3
API Version:  kubevirt.io/v1alpha3
Kind:         VirtualMachine
Metadata:
  Creation Timestamp:  2020-03-02T16:23:30Z
  Generation:          18
  Resource Version:    17376540
  Self Link:           /apis/kubevirt.io/v1alpha3/namespaces/flfr/virtualmachines/debian
  UID:                 ddab2014-e87f-4965-b7ae-3aae4639fa4a
Spec:
  Running:  true
  Template:
    Metadata:
      Creation Timestamp:  <nil>
      Labels:
        kubevirt.io/domain:  debian
        kubevirt.io/size:    small
    Spec:
      Domain:
        Cpu:
          Cores:  2
        Devices:
          Disks:
            Disk:
              Bus:  virtio
            Name:   containervolume
            Disk:
              Bus:  virtio
            Name:   cloudinitvolume
          Interfaces:
            Bridge:
            Name:  default
        Machine:
          Type:  q35
        Resources:
          Requests:
            Memory:  1024M
      Networks:
        Name:  default
        Pod:
      Volumes:
        Container Disk:
          Image:  tedezed/debian-container-disk:9.0
        Name:     containervolume
        Cloud Init No Cloud:
          User Data:  #cloud-config
chpasswd:
  list: |
    debian:debian
    root:toor
  expire: FalseapiVersion: kubevirt.io/v1alpha3
        Name:  cloudinitvolume
Status:
  Created:  true
Events:
  Type    Reason            Age                  From                       Message
  ----    ------            ----                 ----                       -------
  Normal  SuccessfulDelete  34m                  virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 4d64d3af-416c-4d21-9639-d083af774542
  Normal  SuccessfulDelete  29m                  virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 7db99d34-85d7-49f0-89c9-321f18d24910
  Normal  SuccessfulDelete  24m                  virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 953e6077-bd94-4235-b885-ab21498826e9
  Normal  SuccessfulDelete  18m                  virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 34c6fc8d-7f42-4cd3-a0bf-d3c98c541d04
  Normal  SuccessfulDelete  13m                  virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 2afcd195-7adc-429d-920b-02b862d833d5
  Normal  SuccessfulDelete  8m1s                 virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance ad762dec-7ca8-411e-abe8-53bec8987262
  Normal  SuccessfulCreate  2m41s (x8 over 40m)  virtualmachine-controller  Started the virtual machine by creating the new virtual machine instance debian
  Normal  SuccessfulDelete  2m41s                virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance a4612b61-769b-4c3a-8060-d52640fbdf05
```

The log of the virt-launcher pod says:
```
# kubectl logs -f pod/virt-launcher-debian-n8lsd compute
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:68","timestamp":"2020-03-02T16:40:37.524470Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:71","timestamp":"2020-03-02T16:40:37.524712Z"}
{"component":"virt-launcher","level":"info","msg":"Watchdog file created at /var/run/kubevirt/watchdog-files/flfr_debian","pos":"virt-launcher.go:143","timestamp":"2020-03-02T16:40:37.526113Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu:///system","pos":"libvirt.go:355","timestamp":"2020-03-02T16:40:37.537406Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory')","pos":"libvirt.go:363","timestamp":"2020-03-02T16:40:37.537995Z"}
{"component":"virt-launcher","level":"info","msg":"libvirt version: 5.6.0, package: 7.fc31 (Unknown, 2019-11-22-21:16:59, )","subcomponent":"libvirt","thread":"69","timestamp":"2020-03-02T16:40:37.627000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: debian","subcomponent":"libvirt","thread":"69","timestamp":"2020-03-02T16:40:37.627000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get DBus system bus connection: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory","pos":"virDBusGetSystemBus:108","subcomponent":"libvirt","thread":"69","timestamp":"2020-03-02T16:40:37.627000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Child process (/usr/sbin/dmidecode -q -t 0,1,2,3,4,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2792","subcomponent":"libvirt","thread":"69","timestamp":"2020-03-02T16:40:37.633000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:371","timestamp":"2020-03-02T16:40:38.454273Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:294","timestamp":"2020-03-02T16:40:38.458165Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:73","timestamp":"2020-03-02T16:40:38.458374Z"}
panic: timed out waiting for domain to be defined

goroutine 1 [running]:
main.waitForDomainUUID(0x45d964b800, 0xc000544b40, 0xc000cf0180, 0x1a1aee0, 0xc0001348c0, 0xc000544b40)
        cmd/virt-launcher/virt-launcher.go:226 +0x41a
main.main()
        cmd/virt-launcher/virt-launcher.go:434 +0x1031
{"component":"virt-launcher","level":"error","msg":"dirty virt-launcher shutdown","pos":"virt-launcher.go:516","reason":"exit status 2","timestamp":"2020-03-02T16:45:38.470077Z"}
```

I've tried popping a shell in the virt-launcher container and it does not show a great deal of processes:
```
sh-5.0# ps afx
  PID TTY      STAT   TIME COMMAND
  172 pts/0    Ss     0:00 /bin/sh
  197 pts/0    R+     0:00  \_ ps afx
    1 ?        Ssl    0:00 /usr/bin/virt-launcher --qemu-timeout 5m --name debian --uid fe8e454b-75bc-44b0-8976-94d606020a82 --namespace flfr --kubevirt-share-dir /var/run/kubevirt --ephemeral-disk-dir /var/run/kubevirt-ephemeral-disks --container-disk-dir /var/run/kube
   30 ?        Sl     0:00 /usr/bin/virt-launcher --qemu-timeout 5m --name debian --uid fe8e454b-75bc-44b0-8976-94d606020a82 --namespace flfr --kubevirt-share-dir /var/run/kubevirt --ephemeral-disk-dir /var/run/kubevirt-ephemeral-disks --container-disk-dir /var/run/kube
   53 ?        Sl     0:00  \_ /usr/sbin/libvirtd
   54 ?        S      0:00  \_ /usr/sbin/virtlogd -f /etc/libvirt/virtlogd.conf

```

--- a/pkg/virt-handler/container-disk/mount.go
+++ b/pkg/virt-handler/container-disk/mount.go
@@ -41,7 +41,7 @@ func (m *Mounter) Mount(vmi *v1.VirtualMachineInstance, verify bool) error {
 				if err != nil {
 					return fmt.Errorf("failed to detect root mount point of containerDisk %v on the node: %v", volume.Name, err)
 				}
-				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.MountPoint), volume.ContainerDisk.Path)
+				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.Root, nodeMountInfo.MountPoint), volume.ContainerDisk.Path)
 				if err != nil {
 					return fmt.Errorf("failed to find a sourceFile in containerDisk %v: %v", volume.Name, err)
 				}

--- a/pkg/virt-handler/isolation/isolation.go
+++ b/pkg/virt-handler/isolation/isolation.go
@@ -296,7 +296,7 @@ func (r *realIsolationResult) MountInfoRoot() (*MountInfo, error) {
 			}
 		}
 
-		if record[3] == "/" && record[4] == "/" {
+		if record[4] == "/" {
 			return &MountInfo{
 				DeviceContainingFile: record[2],
 				Root:                 record[3],

